Only in harib25c/apilib: api021.nas
Only in harib25c/apilib: api022.nas
Only in harib25c/apilib: api023.nas
Only in harib25c/apilib: api024.nas
Only in harib25c/apilib: api025.nas
Binary files harib25b/apilib/apilib.lib and harib25c/apilib/apilib.lib differ
diff -r harib25b/apilib/Makefile harib25c/apilib/Makefile
4,5c4,5
< 			api019.obj api020.obj \
< 			alloca.obj
---
> 			api019.obj api020.obj api021.obj api022.obj api023.obj api024.obj \
> 			api025.obj alloca.obj
diff -r harib25b/apilib/TRANS.TBL harib25c/apilib/TRANS.TBL
21a22,26
> F API021.NAS;1                                                                                                                                                                                                    	api021.nas
> F API022.NAS;1                                                                                                                                                                                                    	api022.nas
> F API023.NAS;1                                                                                                                                                                                                    	api023.nas
> F API024.NAS;1                                                                                                                                                                                                    	api024.nas
> F API025.NAS;1                                                                                                                                                                                                    	api025.nas
diff -r harib25b/apilib.h harib25c/apilib.h
20a21,25
> int api_fopen(char *fname);
> void api_fclose(int fhandle);
> void api_fseek(int fhandle, int offset, int mode);
> int api_fsize(int fhandle, int mode);
> int api_fread(char *buf, int maxsize, int fhandle);
diff -r harib25b/app_make.txt harib25c/app_make.txt
29c29
< 	-$(OBJ2BIM) @$(RULEFILE) out:$(APP).bim map:$(APP).map stack:$(STACK) \
---
> 	$(OBJ2BIM) @$(RULEFILE) out:$(APP).bim map:$(APP).map stack:$(STACK) \
diff -r harib25b/haribote/bootpack.h harib25c/haribote/bootpack.h
216a217,218
> 	struct FILEHANDLE *fhandle;
> 	int *fat;
249a252,256
> };
> struct FILEHANDLE {
> 	char *buf;
> 	int size;
> 	int pos;
diff -r harib25b/haribote/console.c harib25c/haribote/console.c
12a13
> 	struct FILEHANDLE fhandle[8];
13a15
> 
25a28,32
> 	for (i = 0; i < 8; i++) {
> 		fhandle[i].buf = 0;	/* 未使用マーク */
> 	}
> 	task->fhandle = fhandle;
> 	task->fat = fat;
387a395,400
> 			for (i = 0; i < 8; i++) {	/* クローズしてないファイルをクローズ */
> 				if (task->fhandle[i].buf != 0) {
> 					memman_free_4k(memman, (int) task->fhandle[i].buf, task->fhandle[i].size);
> 					task->fhandle[i].buf = 0;
> 				}
> 			}
413a427,429
> 	struct FILEINFO *finfo;
> 	struct FILEHANDLE *fh;
> 	struct MEMMAN *memman = (struct MEMMAN *) MEMMAN_ADDR;
528a545,601
> 	} else if (edx == 21) {
> 		for (i = 0; i < 8; i++) {
> 			if (task->fhandle[i].buf == 0) {
> 				break;
> 			}
> 		}
> 		fh = &task->fhandle[i];
> 		reg[7] = 0;
> 		if (i < 8) {
> 			finfo = file_search((char *) ebx + ds_base,
> 					(struct FILEINFO *) (ADR_DISKIMG + 0x002600), 224);
> 			if (finfo != 0) {
> 				reg[7] = (int) fh;
> 				fh->buf = (char *) memman_alloc_4k(memman, finfo->size);
> 				fh->size = finfo->size;
> 				fh->pos = 0;
> 				file_loadfile(finfo->clustno, finfo->size, fh->buf, task->fat, (char *) (ADR_DISKIMG + 0x003e00));
> 			}
> 		}
> 	} else if (edx == 22) {
> 		fh = (struct FILEHANDLE *) eax;
> 		memman_free_4k(memman, (int) fh->buf, fh->size);
> 		fh->buf = 0;
> 	} else if (edx == 23) {
> 		fh = (struct FILEHANDLE *) eax;
> 		if (ecx == 0) {
> 			fh->pos = ebx;
> 		} else if (ecx == 1) {
> 			fh->pos += ebx;
> 		} else if (ecx == 2) {
> 			fh->pos = fh->size + ebx;
> 		}
> 		if (fh->pos < 0) {
> 			fh->pos = 0;
> 		}
> 		if (fh->pos > fh->size) {
> 			fh->pos = fh->size;
> 		}
> 	} else if (edx == 24) {
> 		fh = (struct FILEHANDLE *) eax;
> 		if (ecx == 0) {
> 			reg[7] = fh->size;
> 		} else if (ecx == 1) {
> 			reg[7] = fh->pos;
> 		} else if (ecx == 2) {
> 			reg[7] = fh->pos - fh->size;
> 		}
> 	} else if (edx == 25) {
> 		fh = (struct FILEHANDLE *) eax;
> 		for (i = 0; i < ecx; i++) {
> 			if (fh->pos == fh->size) {
> 				break;
> 			}
> 			*((char *) ebx + ds_base + i) = fh->buf[fh->pos];
> 			fh->pos++;
> 		}
> 		reg[7] = i;
Binary files harib25b/haribote/haribote.sys and harib25c/haribote/haribote.sys differ
diff -r harib25b/Makefile harib25c/Makefile
23c23,24
< 		sosu/sosu.hrb sosu2/sosu2.hrb sosu3/sosu3.hrb
---
> 		sosu/sosu.hrb sosu2/sosu2.hrb sosu3/sosu3.hrb \
> 		typeipl/typeipl.hrb
47a49
> 		copy from:typeipl/typeipl.hrb to:@: \
82a85
> 	$(MAKE) -C typeipl
126a130
> 	$(MAKE) -C typeipl		clean
149a154
> 	$(MAKE) -C typeipl		src_only
diff -r harib25b/TRANS.TBL harib25c/TRANS.TBL
22a23
> D TYPEIPL                                                                                                                                                                                                         	typeipl
Only in harib25c: typeipl
